src/app/pages/applications/dialogs/pod-select/pod-select-dialog.component.ts

```diff
@@ -81,27 +95,34 @@ export class PodSelectDialogComponent implements OnInit {
             this.containers$ = of(this.podDetails[value].map((item) => ({ label: item, value: item })));
             this.form.controls.containers.patchValue(this.podDetails[value][0]);
           });
+        } else {
+          this.hasPool = false;
         }
       },
       error: () => {
+        this.hasPool = false;
         this.loader.close();
       },
     });
   }
 
   onPodSelect(): void {
-    const pod = this.form.value.pods;
+    if (this.data.customSubmit) {
+      this.data.customSubmit(this);
```

Comment:
This goes back to our discussion about how entity-form ended up with so many options :)

How about extracting all submit functions to relevant components and just make dialog return something? Or at least have every consumer of dialog provide a submit function.

Reply:
All submit functions extracted to relevant components.